File: apps/api-dotnet/src/AdventureWorks.Models/Features/Sales/SalesOrderAnalyticsModel.cs
namespace AdventureWorks.Models.Features.Sales;

/// <summary>Aggregated analytics for a filtered slice of sales orders.</summary>
public sealed class SalesOrderAnalyticsModel
{
    /// <summary>Sum of TotalDue for all matching orders.</summary>
    public decimal TotalRevenue { get; set; }

    /// <summary>Number of matching orders.</summary>
    public int OrderCount { get; set; }

    /// <summary>Filtered revenue as a percentage (0–100) of the all-time unfiltered revenue.</summary>
    public decimal PercentageOfTotal { get; set; }

    /// <summary>
    /// Monthly revenue trend for the matching orders, ordered oldest-first.
    /// Capped at 24 entries; date ranges spanning more than 24 months drop the oldest months.
    /// </summary>
    public IReadOnlyList<SalesOrderMonthlyTrendModel> MonthlyTrend { get; set; } = [];
}

File: apps/angular-web/libs/sales/data-access/src/lib/models/sales-order-analytics.model.ts
/** Revenue total for a single calendar month within an analytics trend series. */
export interface SalesOrderMonthlyTrend {
  readonly year: number;
  readonly month: number;
  readonly revenue: number;
}

/** Aggregated analytics for a filtered slice of sales orders. */
export interface SalesOrderAnalytics {
  readonly totalRevenue: number;
  readonly orderCount: number;
  /** Filtered revenue as a percentage (0–100) of the all-time unfiltered revenue. */
  readonly percentageOfTotal: number;
  readonly monthlyTrend: SalesOrderMonthlyTrend[];
}

/**
 * Filter-only subset of SalesOrderParams — no pagination fields.
 * `accountNumber` is intentionally absent; the analytics panel does not filter by account.
 */
export interface SalesOrderAnalyticsFilter {
  readonly orderDateFrom?: string;
  readonly orderDateTo?: string;
  readonly status?: number;
  readonly salesPersonId?: number;
  readonly territoryId?: number;
}
